### PR TITLE
Separate external table options out of copy option check

### PR DESCRIPTION
**Pull request by pf-qiu**
ProcessCopyOptions checks the option list of COPY command. It's also
called by external table when text/csv format is used. It's better
not to mix external table specific options here and check them
separately.

Checking custom protocol here is not necessary because it's checked
when parsing location urls in GenerateExtTableEntryOptions anyway.

### CODE REVIEW

**Review by frankgh — COMMENTED** [review]
From a PXF point of view, this seems to be fine. I tested it locally. If we need any additional validation on the PXF side, we can move the logic there.

**Review by JunfengYang — COMMENTED** [review]
Looks good for me except for the issue I mentioned.

File: src/backend/commands/copy.c [hunk 1]
```diff
@@ -1675,12 +1652,6 @@ ProcessCopyOptions(ParseState *pstate,
 						 errmsg("conflicting or redundant options")));
 			cstate->skip_ext_partition = true;
 		}
-		else if (!extprotocol_oid)
-			ereport(ERROR,
```
[the remark this answers is not in the document]
Comment (pf-qiu) [reply src/backend/commands/copy.c:1679]: Ah, shouldn't remove that check. Only "extprotocol_oid".

**Review by JunfengYang — APPROVED** [review]
+1

**Pull request merged**